src/middlewared/middlewared/plugins/apps/ix_apps/query.py

```diff
@@ -85,19 +86,22 @@ def list_apps(
         workloads = translate_resources_to_desired_workflow(app_resources)
         # When we stop docker service and start it again - the containers can be in exited
         # state which means we need to account for this.
-        state = 'STOPPED'
-        exited_containers = 0
+        state = AppState.STOPPED
+        workload_stats = defaultdict(int)
+        workloads_len = len(workloads['container_details'])
         for container in workloads['container_details']:
-            if container['state'] == ContainerState.STARTING.value:
-                state = AppState.DEPLOYING.value
-                break
-            elif container['state'] == ContainerState.RUNNING.value:
-                state = AppState.RUNNING.value
-            elif container['state'] == ContainerState.EXITED.value:
-                exited_containers += 1
-        else:
-            if exited_containers != 0 and exited_containers == len(workloads['container_details']):
-                state = AppState.CRASHED.value
+            workload_stats[container['state']] += 1
+
+        if workload_stats[ContainerState.CRASHED.value]:
+            state = AppState.CRASHED
+        elif workload_stats[ContainerState.CREATED.value] or workload_stats[ContainerState.STARTING.value]:
+            state = AppState.DEPLOYING
+        elif 0 < workloads_len == sum(
+            workload_stats[k.value] for k in (ContainerState.RUNNING, ContainerState.EXITED)
```

Comment:
This will instantiate 2 separate instances of that enum every time this is called. Can we do this a more efficient way?